features/cellular/framework/API/CellularContext.h

```diff
@@ -230,6 +239,17 @@ class CellularContext : public CellularBase {
     */
     virtual void cellular_callback(nsapi_event_t ev, intptr_t ptr) = 0;
 
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN && NSAPI_PPP_AVAILABLE)
```

Comment:
Not sure whether `enable_hup` really needs to be conditional on PPP, but it certainly isn't conditional on UARTSerial, as an abstract API. I'd simplify by making this unconditional.

If it is conditional in any way, you should at least provide a dummy method in the `#else` so callers don't all need ifdef.

(Callers definitely shouldn't have ifdef, because maybe someone has inherited from `AT_CellularDevice` and provided an alternate enable_hup for a totally alien FileHandle).

Reply:
removed #ifdef